server/monitor.go

```diff
@@ -622,14 +649,18 @@ type RouteInfo struct {
 	OutBytes     int64              `json:"out_bytes"`
 	NumSubs      uint32             `json:"subscriptions"`
 	Subs         []string           `json:"subscriptions_list,omitempty"`
+	SubsDetail   []SubDetail        `json:"subscriptions_list_detail,omitempty"`
 }
 
 // Routez returns a Routez struct containing information about routes.
 func (s *Server) Routez(routezOpts *RoutezOptions) (*Routez, error) {
 	rs := &Routez{Routes: []*RouteInfo{}}
 	rs.Now = time.Now()
 
-	subs := routezOpts != nil && routezOpts.Subscriptions
+	opts := RoutezOptions{}
+	if routezOpts != nil {
+		opts = *routezOpts
```

Comment:
I would not do that (although probably safe here). Making a `*routezOpts` could become wrong if options had field that would not be copied but referenced.
Why not something like:
```
var subs, subsDet bool
if routezOpts != nil {
    subs = routezOpts.Subscriptions
    subsDet = routezOpts.SubscriptionsDetail
}
```
and then use subs/subsDet below in your code.

Reply:
You either missed this comment or decided not to change, but I think making a copy of the options is not the best approach..